bluemira/power_cycle/net.py

```diff
@@ -0,0 +1,644 @@
+        """Get the config"""
+        if obj is None:
+            return list
+        return getattr(obj, self._name)
+
+    def __set__(
+        self,
+        obj: Any,
+        value: Union[Callable[[], list], List[Union[EfficiencyDictType, Efficiency]]],
+    ):
+        """Setup the config"""
+        if callable(value):
+            value = value()
+        for no, v in enumerate(value):
+            if not isinstance(v, Efficiency):
+                value[no] = Efficiency(**v)
+
+        setattr(obj, self._name, value)
+
+
+@dataclass
+class ScenarioConfig(Config):
+    """Power cycle scenario config"""
+
+    pulses: dict[str, int]
+    description: str = ""
+
+
+@dataclass
+class PulseConfig(Config):
+    """Power cycle pulse config"""
+
+    phases: list[str]
+    description: str = ""
+
+
+@dataclass
+class PhaseConfig(Config):
+    """Power cycle phase config"""
+
+    operation: str
+    subphases: list[str]
+    description: str = ""
+
+
+@dataclass
+class SubPhaseConfig(Config):
+    """SubPhase Config"""
+
+    duration: Union[float, str]
+    loads: list[str] = field(default_factory=list)
+    efficiencies: PhaseEfficiencyDescriptor = PhaseEfficiencyDescriptor()
+    unit: str = "s"
+    description: str = ""
+    reference: str = ""
+
+    def __post_init__(self):
+        """Enforce unit conversion"""
+        if isinstance(self.duration, (float, int)):
+            self.duration = raw_uc(self.duration, self.unit, "second")
+            self.unit = "s"
+
+
+@dataclass
+class SystemConfig(Config):
+    """Power cycle system config"""
+
+    subsystems: List[str]
+    description: str = ""
+
+
+@dataclass
+class SubSystemConfig(Config):
+    """Power cycle sub system config"""
+
+    loads: List[str]
+    description: str = ""
+
+
+@dataclass
+class LoadConfig(Config):
+    """Power cycle load config"""
+
+    time: npt.ArrayLike = field(default_factory=lambda: np.arange(2))
+    reactive_data: Optional[npt.ArrayLike] = None
+    active_data: Optional[npt.ArrayLike] = None
+    efficiencies: LoadEfficiencyDescriptor = LoadEfficiencyDescriptor()
+    model: Union[LoadModel, str] = LoadModel.RAMP
+    unit: str = "W"
+    description: str = ""
+    normalised: bool = True
+    consumption: bool = True
+
+    def __post_init__(self):
+        """Validate load"""
+        if self.reactive_data is None:
+            self.reactive_data = np.zeros_like(self.time)
+        if self.active_data is None:
+            self.active_data = np.zeros_like(self.time)
+
+        for var_name in ("time", "reactive_data", "active_data"):
+            var = getattr(self, var_name)
+            if not isinstance(var, np.ndarray):
+                setattr(self, var_name, np.asarray(var))
+        if isinstance(self.model, str):
+            self.model = LoadModel[self.model.upper()]
+        for data in (self.reactive_data, self.active_data):
+            if data.size != self.time.size:
```

Comment:
Should len() be used here instead of .size?

Reply:
no because that wont catch if for whatever reason data is a 2d array, size will tell me the total number of elements not the length of the first axis.